### PR TITLE
Update dependencies, setup, and README (installation etc)

### PR DESCRIPTION
**Pull request by haipinglu**
Fixes [dependencies card](https://github.com/pykale/pykale/projects/9#card-52913368), discussion #20, issues #23, #39

### Description
Update README and `setup.py`

- Removed the flowchart in README to be more compact and reduce redundancy.
- Updated installation instructions in README.
- Keep dependencies (requirements) in `setup.py`, including core dependencies for kale API, and additional/extra dependencies* for examples and development.
- Restructured and updated `setup.py` to be more compact.
- Not all packages in `setup.py` have a min-version specified, which is not uncommon in all popular packages studied. We can keep them like that and specify a min-version when necessary (e.g. errors). Some will report error if an incorrect version is specified.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Review by RaivoKoot — APPROVED** [review]
Nice!

**Comment by haipinglu** [comment]
@bobturneruk Would you be able to provide your review? If any question, we can discuss tomorrow. Many thanks.

**Comment by bobturneruk** [comment]
I'll have a look at it.

**Comment by bobturneruk** [comment]
I'm struggling to install (Windows 10, conda):
```
(pykale) PS C:\Users\Bob Turner\Documents\pykale> conda create --force -n pykale python=3.6
WARNING: A conda environment already exists at 'C:\Users\Bob Turner\.conda\envs\pykale'
Remove existing environment (y/[n])? y

Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\Bob Turner\.conda\envs\pykale

  added / updated specs:
    - python=3.6

The following NEW packages will be INSTALLED:

  certifi            conda-forge/win-64::certifi-2020.12.5-py36ha15d459_1
  pip                conda-forge/noarch::pip-21.0.1-pyhd8ed1ab_0
  python             conda-forge/win-64::python-3.6.12-h39d44d4_0_cpython
  python_abi         conda-forge/win-64::python_abi-3.6-1_cp36m
  setuptools         conda-forge/win-64::setuptools-49.6.0-py36ha15d459_3
  vc                 conda-forge/win-64::vc-14.2-hb210afc_2
  vs2015_runtime     conda-forge/win-64::vs2015_runtime-14.28.29325-h5e1d092_0
  wheel              conda-forge/noarch::wheel-0.36.2-pyhd3deb0d_0
  wincertstore       conda-forge/win-64::wincertstore-0.2-py36ha15d459_1006


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate pykale
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(pykale) PS C:\Users\Bob Turner\Documents\pykale> pip install -e .[extras]
Obtaining file:///C:/Users/Bob%20Turner/Documents/pykale
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting pytorch-lightning
  Downloading pytorch_lightning-1.1.7-py3-none-any.whl (695 kB)
     |████████████████████████████████| 695 kB 2.2 MB/s
Collecting scikit-learn
  Using cached scikit_learn-0.24.1-cp36-cp36m-win_amd64.whl (6.8 MB)
Collecting torch>=1.7.0
  Using cached torch-1.7.1-cp36-cp36m-win_amd64.whl (184.1 MB)
Collecting scikit-image
  Downloading scikit_image-0.17.2-cp36-cp36m-win_amd64.whl (11.5 MB)
     |████████████████████████████████| 11.5 MB 3.3 MB/s
Collecting tensorly
  Using cached tensorly-0.5.1.tar.gz (106 kB)
Collecting numpy>=1.18.0
  Using cached numpy-1.19.5-cp36-cp36m-win_amd64.whl (13.2 MB)
ERROR: Could not find a version that satisfies the requirement torchvision>=0.8.1 (from pykale[extras])
ERROR: No matching distribution found for torchvision>=0.8.1
```

**Comment by haipinglu** [comment]
> pip install -e .[extras]

Thanks. I've fixed it by removing the version requirement for torchvision, which reads version from a non-standard location, I think. It should be fine, as in MONAI.  I tested in conda and the installation worked. Can you test again?

**Comment by bobturneruk** [comment]
I'll check again before we meet.

**Review by bobturneruk — APPROVED** [review]
Looks good. There are a few different ways of doing this and this is a good approach.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,8 +21,7 @@ jobs:
         python-version: "3.6"
     - name: Install dependencies
       run: |
-        pip install pytest nbval jupyter tqdm matplotlib torchvision scipy
-        python setup.py build develop
-    # - name: Run example notebooks
+        pip install -e .[extra]
```
[the remark this answers is not in the document]
Comment (haipinglu) [reply .github/workflows/build.yml:24]: Thanks for your eagle eyes. Indeed, I checked the CI log and actually there was a warning `WARNING: pykale 0.1.0a1 does not provide the extra 'extra'`.

**Pull request merged**